iexec-sms-library/src/main/java/com/iexec/sms/SmsClient.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.sms;
+
+import feign.Headers;
+import feign.Param;
+import feign.RequestLine;
+
+/**
+ * Interface allowing to instantiate a Feign client targeting SMS REST endpoints.
+ * <p>
+ * To create the client, call:
+ * <pre>FeignBuilder.createBuilder(feignLogLevel)
+ *         .target(SmsClient.class, smsUrl)</pre>
+ * @see com.iexec.common.utils.FeignBuilder
+ */
+public interface SmsClient {
```

Comment:
I think we talk about exposing an `SmsService` too, as a default/go-to class for connecting to the SMS, class which can be used or not by a consumer. This would add the benefit of removing some code duplication in consumers.

Reply:
I agree. The idea was to start with the Feign client and to start the refactoring.
I think the service aspect can be done in a second time.
This second batch of PRs could be dispatched between different persons within the team.

Reply:
I fully agree, just wanted to make sure :+1:

Reply:
Hopping in the discussion!
How about renaming this class to `SmsApi` and providing (in the next PR) a new class `SmsClient` that contains some logic for calling the API.